### DISCUSSION TITLE
Shared camera support with ARSceneView?

### DISCUSSION THREAD

**Discussion by carloshwa** [sceneview/sceneview#554] [Q&A]
Is it possible to use a shared camera with `ARSceneView`, using the `Session.Feature.SHARED_CAMERA` session feature as documented here?

https://developers.google.com/ar/develop/java/camera-sharing

https://github.com/google-ar/arcore-android-sdk/blob/main/samples/shared_camera_java/app/src/main/java/com/google/ar/core/examples/java/sharedcamera/SharedCameraActivity.java

**Comment by ThomasGorisse** [comment]
ARCore's Shared Camera mode allows simultaneous access to the camera for custom image processing. In v4.0.0-rc.1, you can access camera frames through the session:

```kotlin
ARSceneView(
    onSessionUpdated = { session, frame ->
        // Access the camera image
        val image = frame.acquireCameraImage()
        // Process the image (ML, CV, etc.)
        image.close()
    }
)
```

For full Shared Camera mode (Camera2 interop), you'd configure the AR session with shared camera access. This is an advanced ARCore feature — SceneView provides the session access and you use ARCore's shared camera API directly.
